app/src/test/java/ar/com/wolox/android/example/ui/login/LoginPresenterTest.kt

```diff
@@ -0,0 +1,80 @@
+package ar.com.wolox.android.example.ui.login
+
+import ar.com.wolox.android.example.model.LoginRequest
+import ar.com.wolox.android.example.model.LoginResponse
+import ar.com.wolox.android.example.network.repository.PostRepository
+import ar.com.wolox.wolmo.core.tests.CoroutineTestRule
+import ar.com.wolox.wolmo.core.tests.WolmoPresenterTest
+import ar.com.wolox.wolmo.networking.retrofit.handler.NetworkResponse
+import com.nhaarman.mockitokotlin2.doReturn
+import com.nhaarman.mockitokotlin2.times
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.runBlocking
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mock
+import org.mockito.Mockito.mock
+import retrofit2.Response
+
+@ExperimentalCoroutinesApi
+class LoginPresenterTest : WolmoPresenterTest<LoginView, LoginPresenter>() {
+
+    @get:Rule
+    val coroutineTestRule = CoroutineTestRule(runOnAllTests = true)
+
+    @Mock
+    lateinit var postRepository: PostRepository
+
+    override fun getPresenterInstance() = LoginPresenter(postRepository)
+
+    @Test
+    fun `given a empty email then show error, enter email`() = runBlocking {
```

Comment:
What about `given an empty email, when the user presses the login button, then an error is shown`? Try yo apply that format to every sentence

Reply:
sorry, I don't understand...